server/routes/api/auth.ts

```diff
@@ -75,13 +76,11 @@ router.post("auth.config", async (ctx) => {
 
   // If subdomain signin page then we return minimal team details to allow
   // for a custom screen showing only relevant signin options for that team.
-  if (
+  else if (
     process.env.SUBDOMAINS_ENABLED === "true" &&
-    isCustomSubdomain(ctx.request.hostname) &&
-    !isCustomDomain(ctx.request.hostname)
+    isHostedSubdomain(ctx.request.hostname)
   ) {
-    const domain = parseDomain(ctx.request.hostname);
-    const subdomain = domain ? domain.subdomain : undefined;
```

Comment:
is the `undefined` here explicitly necessary? this seems like it's very purposeful but if `isCustomSubdomain` comes back as true, this will never be empty